File: DataStructure/387. First Unique Character in a String.cs
public class Solution {
    public int FirstUniqChar(string s) {
        Dictionary<char, int> temp = new Dictionary<char, int>();
        
        for (int i = 0; i < s.Length; i++)
        {
            if (temp.ContainsKey(s[i]))
            {
                temp[s[i]] += 1;
            }
            else 
            {
                temp.Add(s[i], 1);
            }
        }
        
        foreach (char alpha in s)
        {
            if (temp[alpha] == 1)
            {
                return s.IndexOf(alpha);
            }
            
        }
        return -1;
    }
}

File: DataStructure/387. First Unique Character in a String.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        temp = {}   
        for string in s :
            if string not in temp:
                temp[string] = 1
            else :
                temp[string] += 1
        
        for char in temp:
            if temp[char] == 1:
                return s.index(char)
        return -1
        